File: Copilot/Python/Factory_Method/round1/source/menu_factory.py
from abc import ABC, abstractmethod
from beverages import Espresso, Latte, Milkshakes, FruitJuices
from pastries import Croissant, ChocolateCake, ChocolateChipCookies
from snacks import TunaSandwich, Spaghetti, VegetableSalad

class MenuFactory(ABC):
    @abstractmethod
    def create_menu_item(self, type):
        pass

class BeverageFactory(MenuFactory):
    def create_menu_item(self, type):
        if type == "Espresso":
            return Espresso()
        elif type == "Latte":
            return Latte()
        elif type == "Milkshakes":
            return Milkshakes()
        elif type == "Fruit Juices":
            return FruitJuices()
        else:
            raise ValueError("Unknown beverage type")

class PastryFactory(MenuFactory):
    def create_menu_item(self, type):
        if type == "Croissant":
            return Croissant()
        elif type == "Chocolate Cake":
            return ChocolateCake()
        elif type == "Chocolate Chip Cookies":
            return ChocolateChipCookies()
        else:
            raise ValueError("Unknown pastry type")

class SnackFactory(MenuFactory):
    def create_menu_item(self, type):
        if type == "Tuna Sandwich":
            return TunaSandwich()
        elif type == "Spaghetti":
            return Spaghetti()
        elif type == "Vegetable Salad":
            return VegetableSalad()
        else:
            raise ValueError("Unknown snack type")

File: Copilot/Java/Factory_Method/round2/source/MenuFactory.java
public abstract class MenuFactory {
    public abstract MenuItem createMenuItem(String type);
}

public class BeverageFactory extends MenuFactory {
    @Override
    public MenuItem createMenuItem(String type) {
        switch (type) {
            case "Espresso":
                return new Espresso();
            case "Latte":
                return new Latte();
            case "Milkshakes":
                return new Milkshakes();
            case "Fruit Juices":
                return new FruitJuices();
            default:
                throw new IllegalArgumentException("Unknown beverage type");
        }
    }
}

public class PastryFactory extends MenuFactory {
    @Override
    public MenuItem createMenuItem(String type) {
        switch (type) {
            case "Croissant":
                return new Croissant();
            case "Chocolate Cake":
                return new ChocolateCake();
            case "Chocolate Chip Cookies":
                return new ChocolateChipCookies();
            default:
                throw new IllegalArgumentException("Unknown pastry type");
        }
    }
}

public class SnackFactory extends MenuFactory {
    @Override
    public MenuItem createMenuItem(String type) {
        switch (type) {
            case "Tuna Sandwich":
                return new TunaSandwich();
            case "Spaghetti":
                return new Spaghetti();
            case "Vegetable Salad":
                return new VegetableSalad();
            default:
                throw new IllegalArgumentException("Unknown snack type");
        }
    }
}